contributors/design-proposals/local-storage-overview.md

```diff
@@ -0,0 +1,380 @@
+          containers:
+          - name: nginx
+            image: gcr.io/google_containers/nginx-slim:0.8
+            ports:
+            - containerPort: 80
+              name: web
+            volumeMounts:
+            - name: www
+              mountPath: /usr/share/nginx/html
+            - name: log
+              mountPath: /var/log/nginx
+      volumeClaimTemplates:
+      - metadata:
+          name: www
+          labels:
+            storage.kubernetes.io/medium: ssd
+        spec:
+          volume-type: local
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 100Gi
+      - metadata:
+          name: log
+          labels:
+            storage.kubernetes.io/medium: hdd
+        spec:
+          volume-type: local
+          accessModes: [ "ReadWriteOnce" ]
+          resources:
+            requests:
+              storage: 1Gi
+    ```
+
+5. The scheduler identifies nodes for each pods that can satisfy cpu, memory, storage requirements and also contains free local PVs to satisfy the pods PVC claims. It then binds the pod’s PVCs to specific PVs on the node and then binds the pod to the node. 
+    ```
+    $ kubectl get pvc
+    NAME            STATUS VOLUME     CAPACITY ACCESSMODES … NODE  
+    www-local-pvc-1 Bound  local-pv-1 100Gi    RWO           node-1
+    www-local-pvc-2 Bound  local-pv-1 100Gi    RWO           node-2
+    www-local-pvc-3 Bound  local-pv-1 100Gi    RWO           node-3
+    log-local-pvc-1 Bound  local-pv-2 10Gi     RWO           node-1
+    log-local-pvc-2 Bound  local-pv-2 10Gi     RWO           node-2
+    log-local-pvc-3 Bound  local-pv-2 10Gi     RWO           node-3
+    ```
+    ```
+    $ kubectl get pv
+    NAME       CAPACITY … STATUS    CLAIM           NODE  
+    local-pv-1 100Gi      Bound     www-local-pvc-1 node-1
+    local-pv-2 10Gi       Bound     log-local-pvc-1 node-1
+    local-pv-1 100Gi      Bound     www-local-pvc-2 node-2
+    local-pv-2 10Gi       Bound     log-local-pvc-2 node-2
+    local-pv-1 100Gi      Bound     www-local-pvc-3 node-3
+    local-pv-2 10Gi       Bound     log-local-pvc-3 node-3
+    ```
+
+6. If a pod dies and is replaced by a new one that reuses existing PVCs, the pod will be placed on the same node where the corresponding PVs exist. Stateful Pods are expected to have a high enough priority which will result in such pods preempting other low priority pods if necessary to run on a specific node.
+7. If a new pod fails to get scheduled while attempting to reuse an old PVC, the StatefulSet controller is expected to give up on the old PVC (delete & recycle) and instead create a new PVC based on some policy. This is to guarantee scheduling of stateful pods.
+8. If a PV gets tainted as unhealthy, the StatefulSet is expected to delete pods if they cannot tolerate PV failures. Unhealthy PVs once released will not be added back to the cluster until the corresponding local storage device is healthy.
```

Comment:
Sorry if I was not clear. I am saying the opposite. The "application monitoring the storage device" referred to above is based on the design statement that " Health is monitored by an external entity like the “Node Problem Detector” which is expected to place appropriate taints." 
Rather than have every controller/operator attempt to heuristically guess when it should delete a PVC, might it not be better to have the "Node Problem Detector", kublet, or another local agent make a decision that the volume is no longer usable due to device failure, and to set the associated PVC back to pending. Perhaps using your suggestion above to retrain the volume for data recovery purposes. I can't think of a distributed storage application that will want to re-balance or re-replicate its data due to a temporary network partition or intermittent node failure. The only time, IMO, that we'd want a controller/operator to move a Pod with local PVs to a new node is if the storage device failed, or if the MTTR is so high that it might as well have. In the former case it might be best if a node local agent made the decision that the storage device is failed. In the latter case, we should at least consider having a global controller with a policy that can unbind local PVs from PVCs, rather than having every controller/operator have to implement its own policy.